### PR TITLE
1417 add participants automatically

### PR DESCRIPTION
**Pull request by gbochora**
## Summary
Add the ability to automatically add participants to the channel when inviting them.

https://user-images.githubusercontent.com/4368372/198704724-af0e18e0-c21d-4fd1-a0ac-0b7e9c53c713.mp4

## Ticket Link
https://github.com/mattermost/mattermost-plugin-playbooks/issues/1417

## Checklist

~~- [ ] Telemetry updated~~
~~- [ ] Gated by experimental feature flag~~
- [x] Unit tests updated

### CODE REVIEW

File: webapp/src/components/backstage/playbook_runs/playbook_run/add_participant_modal.tsx [hunk 1]
```diff
@@ -32,8 +34,10 @@ const AddParticipantsModal = ({playbookRun, id, title, show, hideModal}: Props)
     const dispatch = useDispatch();
     const [profiles, setProfiles] = useState<UserProfile[]>([]);
     const {addToRun} = useManageRunMembership(playbookRun.id);
-    const [addToChannel, setAddToChannel] = useState(false);
+    const [forceAddToChannel, setForceAddToChannel] = useState(false);
+    const [channel, meta] = useChannel(playbookRun.channel_id);
     const isChannelMember = useSelector(isCurrentUserChannelMember(playbookRun.channel_id));
+    const isPrivateChannelWithAccess = meta.error === null && channel?.type === General.PRIVATE_CHANNEL;
```
Comment (gbochora) [inline webapp/src/components/backstage/playbook_runs/playbook_run/add_participant_modal.tsx:40]: `isChannelMember` is only valid when the channel is public; we don't have membership information for private channels. So, I'm checking if the current user can access the channel to handle the private channel case.

File: webapp/src/graphql/runs.graphql [hunk 2]
```diff
@@ -12,8 +12,8 @@ mutation UpdateRun($id: String!, $updates: RunUpdates!) {
   updateRun(id: $id, updates: $updates)
 }
 
-mutation AddRunParticipants($runID: String!, $userIDs: [String!]!) {
-	addRunParticipants(runID: $runID, userIDs: $userIDs)
+mutation AddRunParticipants($runID: String!, $userIDs: [String!]!, $forceAddToChannel: Boolean = false) {
```
Comment (trilopin) [inline webapp/src/graphql/runs.graphql:15]: if here is already defaulted to false, do we need to handle in the root resolver as a *bool? Could we simplify root resolved setting it as bool?

File: tests-e2e/cypress/integration/runs/rdp_rhs_participants_spec.js [hunk 3]
```diff
@@ -122,25 +122,104 @@ describe('runs > run details page > rhs > participants', () => {
...
+                    cy.get(`#postMessageText_${id}`).within(() => {
+                        // just to wait until the username is fetched
+                        cy.contains('Someone').should('not.exist');
+                        cy.contains('You were added to the channel by @playbooks.');
+                        cy.contains(`@${testViewerUser.username}`).should('not.exist');
+                    });
+                });
+            });
+
+            it('join action disabled, checkbox selected', () => {
```
Comment (trilopin) [inline tests-e2e/cypress/integration/runs/rdp_rhs_participants_spec.js:189]: this tests is failing when I run locally

**Review by trilopin — APPROVED** [review]
Code is fine and feature works!

Some non-blocking points:
- should we carry in the telemetry whether addParticipants imply channel members or not (taking into account the 3 possible values no/action/forced). Maybe this is more for your effort @crspeller 
- could we simplify root resolver making force flag not nullable (it already has a default false value)

File: webapp/src/graphql/runs.graphql (hunk 2 above)
Comment (gbochora) [reply webapp/src/graphql/runs.graphql:15]: I've tried that, and graphql fails to initialize:
`"error": "field "ForceAddToChannel": bool is not a pointer or a nullable type.`

I'll try one more time, maybe I'm missing something.

**Review by abhijit-singh — APPROVED** [review]
Looks good!

**Comment by abhijit-singh** [comment]
@gbochora One small non-blocking request— Can we make the font-weight of the checkbox text "_Also add people to the channel linked to this run_" `normal` or `400` instead of the current `700`?

**Review by crspeller — APPROVED** [review]

**Pull request merged**